src/Components/App/App.tsx

```diff
@@ -44,23 +46,42 @@ const App: FC = () => {
 
 
   return (
+
     <div className="App">
-      <Header />
-      <main className='main'>
-        <Form onSearch={addCity}/>
-        {error ? <div className={"error"}>{error}</div> : null}
-        {warning ? <div className={"warning"}>{warning}</div> : null}
-
-        {!cities.length ? <h2 className='no-city'>No city forecasts to show</h2> : <HomeCityCards
-          allCities={cities}
-          onSelect={city => {
-            setCurrentCity(city)
-            setDetails(city)
-          }}
-          current={currentCity}
+    <Route exact path="/" render={() => {
+      return (
+        <>
+        <Header />
+        <main className='main'>
+          <Form onSearch={addCity}/>
+          {error ? <div className={"error"}>{error}</div> : null}
+          {warning ? <div className={"warning"}>{warning}</div> : null}
+
+          {!cities.length ? <h2 className='no-city'>No city forecasts to show</h2> : <HomeCityCards
+            allCities={cities}
+            onSelect={city => {
+              setCurrentCity(city)
+              setDetails(city)
+            }}
+            clickedCard={currentCity}
+            details={forecastDetails}
+          />}
+        </main>
+        </>
+      )
+    }} />
+    <Route exact path="/:name" render={({ match }) => {
```

Comment:
This route works perfectly but if we just do localhost:3000/cityname we run into an error now.